src/live.rs

```diff
@@ -63,34 +63,37 @@ pub fn iso_ignition_embed(config: &IsoIgnitionEmbedConfig) -> Result<()> {
         }
     };
 
-    let cpio = make_cpio(&ignition)?;
-    if cpio.len() > embed.length {
-        bail!(
-            "Compressed Ignition config is too large: {} > {}",
-            cpio.len(),
-            embed.length
-        );
-    }
-    if !config.force {
-        // Ensure all zero bytes
-        embed.seek_to_start()?;
-        let mut buf = embed.new_buffer();
-        embed.read(&mut buf)?;
-        // compare to zeroed buffer
-        if buf != embed.new_buffer() {
-            bail!("This ISO image already has an embedded Ignition config; use -f to force.");
-        }
-    }
-    // delete any existing config
-    embed.clear()?;
-    // write new config
-    embed.seek_to_start()?;
-    embed.write(&cpio)?;
-    if config.output.as_deref() == Some("-") {
+    if config.output.as_deref() == USE_STDOUT {
+        let cpio = make_cpio(&ignition)?;
         stdout().write_all(&cpio).chain_err(|| "writing output")?;
```

Comment:
This isn't embedding the CPIO in the ISO image, it's just generating the CPIO and sending it directly to stdout.  `coreos-installer pxe ignition wrap` already does that, if desired.

Reply:
whoosh. it didn't feel right when I was doing it but I went for it anyway.